### PR TITLE
add sentry integration for error-handling

### PR DESCRIPTION
**Pull request by dswah**
- refactor library to build main fastapi app using user defined arguments (like sentry auth token)
- add integrations/sentry.py with sentry-specific error logic and cli options
- build modular cli parser for integration options
- modify sidecar.py and serve.py to accept new integration cli options and build fastapi app accordingly
- add package changes to poetry

### CODE REVIEW

File: meowlflow/api/middlewares/errors.py [hunk 1]
```diff
@@ -9,25 +9,48 @@
...
+                    logger.error(handler_error)
+                    logger.error(traceback.format_exc())
+
+            return JSONResponse(
+                {"error": error.to_dict()},
+                status_code=error.status_code,
+            )
+
+        # now try to catch all unexpected Exceptions
+        except Exception as error:  # pylint: disable=broad-except
```
Comment (dswah) [inline meowlflow/api/middlewares/errors.py:36]: the downside of capturing errors this way is that our integrations (like sentry)  believe them to be "handled"
image

Comment (dswah) [reply meowlflow/api/middlewares/errors.py:36]: ALSO!
the current `sentry.handler` causes all local variables to be lost, and all of the good automatic sentry tracing is not included.

Comment (squat) [reply meowlflow/api/middlewares/errors.py:36]: Wait, that seems like a bad tradeoff, if we don't get any of the variables or tracing!

File: meowlflow/integrations/sentry.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import click
+import sentry_sdk as sentry_sdk
```
Comment (squat) [inline meowlflow/integrations/sentry.py:2]: importing with `as` to explicitly re-export the package

File: examples/document_splitter_schema.json [hunk 3]
```diff
@@ -1 +1 @@
...
...cription": "Successful Response", "content": {"application/json": {"schema": {"$ref": "#/components/schemas/Response"}}}}, "422": {"description": "Validation Error", "content": {"application/json": {"schema": {"$ref": "#/components/schemas/HTTPValidationError"}}}}}}}}, "components": {"schemas": {"HTTPValidationError": {"title": "HTTPValidationError", "type": "object", "properties": {"detail": {"title": "Detail", "type": "array", "items": {"$ref": "#/components/schemas/ValidationError"}}}}, "Request": {"title": "Request", "type": "array", "items": {"type": "string"}, "example": ["page 1 of 2\nfoo", "page 2 of 2\nbar", "page 1 of 1\nbaz"]}, "Response": {"title": "Response", "required": ["predictions"], "type": "object", "properties": {"predictions": {"title": "Predictions", "type": "array", "items": {"type": "integer"}}}, "example": {"predictions": [1, 0, 1]}}, "ValidationError": {"title": "ValidationError", "required": ["loc", "msg", "type"], "type": "object", "properties": {"loc": {"title": "Location", "type": "array", "items": {"anyOf": [{"type": "string"}, {"type": "integer"}]}}, "msg": {"title": "Message", "type": "string"}, "type": {"title": "Error Type", "type": "string"}}}}}}
```
Comment (squat) [inline examples/document_splitter_schema.json:1]: The diff here was:
```
<               "type": "string"
---
>               "anyOf": [
>                 {
>                   "type": "string"
>                 },
>                 {
>                   "type": "integer"
>                 }
>               ]
```

File: meowlflow/integrations/sentry.py (hunk 2 above)
Comment (dswah) [reply meowlflow/integrations/sentry.py:2]: wat!!!

**Review by squat — APPROVED** [review]
🎉

**Pull request merged**